### ISSUE TITLE
How can we extract the features from the trained models?

### ISSUE THREAD

**Issue by livst** [yjxiong/caffe#116]
I couldn't find any example of feature extraction for this repo. Is there any way to do it? If yes, how should I do it?

Thank you!

**Comment by yjxiong** [comment]
You can use the standard ways of Caffe to extract features.

See

http://caffe.berkeleyvision.org/gathered/examples/feature_extraction.html

or

http://nbviewer.jupyter.org/github/BVLC/caffe/blob/master/examples/00-classification.ipynb

**Comment by livst** [comment]
Well, I already tried it and got the following error:

Unknown database backend.

**Comment by livst** [comment]
I've managed to extract the features but there's a problem. For example, if a use the RGB split, i get 1 feature vector / video, instead getting 1 feature vector for every frame of that video. Also I don't get the same number of features for every instance.

So how can i extract the features from the model?

**Comment by livst** [comment]
I've managed to extract the features following your first link. I used the vgg_16_rgb_train_val_fast.prototxt with the training phase input train_rgb_split1.txt. 

The output of the extract_features.bin is only 1 feat vector / video. 

So, for example, i get only 1 features vector for this file:
ucf101_rgb_img/WallPushups/v_WallPushups_g21_c06 111 98

Shouldn't I get a features vector for every frame of that video file?

So if i use the vgg_16_flow_train_val_fast.prototxt, i should get for that video file a feature vector for each pairs of 10 optical flow images ( x and y).

The problem is that i get only 1 features vector for the whole folder.

So either there is an error somewhere, or the features extraction tool extract the features for the whole video file and then it takes the mean of the feat vectors, but i don't see that in the code.
